compiler/src/parsing/lexer.re

```diff
@@ -108,7 +108,8 @@ let dec_float_integral = [%sedlex.regexp?
 let dec_float = [%sedlex.regexp?
   (dec_float_integral, dec_float_decimal, Opt(dec_float_exp)) |
   (dec_float_decimal_explicit, Opt(dec_float_exp)) |
-  (dec_float_integral, dec_float_exp)
+  (dec_float_integral, dec_float_exp) |
+  ("Infinity" | "NaN")
```

Comment:
Just to make this slightly cleaner, I'd make a `dec_float_alphabetic` that's just this, and then reference it here.